ion-c-sys/src/lib.rs

```diff
@@ -2,166 +2,127 @@
-//! }
+//! ionc!(ion_reader_next(ion_reader, &mut ion_type))?;
 //! assert_eq!(ion_type as u32, tid_STRUCT_INT);
 //!
 //! // step into the struct
-//! unsafe {
-//!     let err = ion_reader_step_in(ion_reader);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//! }
+//! ionc!(ion_reader_step_in(ion_reader))?;
 //!
 //! // step to the field
-//! unsafe {
-//!     let err = ion_reader_next(ion_reader, &mut ion_type);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//! }
+//! ionc!(ion_reader_next(ion_reader, &mut ion_type))?;
 //! assert_eq!(ion_type as u32, tid_INT_INT);
 //!
 //! // retrieve the field name--which is 'borrowed' while we don't move the reader
-//! let mut ion_str: ION_STRING = Default::default();
-//! unsafe {
-//!     let err = ion_reader_get_field_name(ion_reader, &mut ion_str);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//!
-//!     let field_name = str::from_utf8(
-//!         slice::from_raw_parts(ion_str.value, ion_str.length as usize)
-//!     ).unwrap();
-//!     assert_eq!(field_name, "a");
-//! }
+//! let mut ion_str = ION_STRING::default();
+//! ionc!(ion_reader_get_field_name(ion_reader, &mut ion_str))?;
+//! assert_eq!(ion_str.as_str()?, "a");
 //!
 //! // read the integer value
 //! let mut int_value: i64 = 0;
-//! unsafe {
-//!     let err = ion_reader_read_int64(ion_reader, &mut int_value);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//! }
+//! ionc!(ion_reader_read_int64(ion_reader, &mut int_value))?;
 //! assert_eq!(int_value, 2);
 //!
 //! // step to the end of the struct
-//! unsafe {
-//!     let err = ion_reader_next(ion_reader, &mut ion_type);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//! }
+//! ionc!(ion_reader_next(ion_reader, &mut ion_type))?;
 //! assert_eq!(ion_type as i32, tid_EOF_INT);
 //!
 //! // step out of the struct
-//! unsafe {
-//!     let err = ion_reader_step_out(ion_reader);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//! }
+//! ionc!(ion_reader_step_out(ion_reader))?;
 //!
 //! // step to the end of the stream
-//! unsafe {
-//!     let err = ion_reader_next(ion_reader, &mut ion_type);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//! }
+//! ionc!(ion_reader_next(ion_reader, &mut ion_type))?;
 //! assert_eq!(ion_type as i32, tid_EOF_INT);
 //!
 //! // close the reader
-//! unsafe {
-//!     let err = ion_reader_close(ion_reader);
-//!     assert_eq!(err, ion_error_code_IERR_OK);
-//! }
+//! ionc!(ion_reader_close(ion_reader));
+//!
+//! # Ok(())
+//! # }
 //! ```
 //!
 //! ### Ion Writer
 //! Here is an end-to-end example of writing some Ion data.
 //!
 //! ```
 //! use std::ptr;
+//!
 //! use ion_c_sys::*;
+//! use ion_c_sys::result::*;
+//! # fn main() -> IonCResult {
 //!
 //! // output buffer
 //! let mut buf: Vec<u8> = vec![0; 128];
 //!
 //! // writer options--emit binary
-//! let mut writer_options: ION_WRITER_OPTIONS = Default::default();
+//! let mut writer_options = ION_WRITER_OPTIONS::default();
```

Comment:
did we add `#[derive(Default)]` for `ION_WRITER_OPTIONS` in another PR?

Reply:
Yes, I added it in the doc test PR #39 because I used that there.  It is actually a feature of `bindgen`.